### DISCUSSION TITLE
How do I embed this.

### DISCUSSION THREAD

**Discussion by 1vers1on** [copy/v86#803] [Q&A]
I dont know how to embed this. my code is
```
<!doctype html>
<script src="./build/libv86.js"></script>
<div id="my_console"></div>
<script>
"use strict";

window.onload = function()
{
    var emulator = window.emulator = new V86Starter({
        wasm_path: "./build/v86.wasm",
        memory_size: 32 * 1024 * 1024,
        vga_memory_size: 2 * 1024 * 1024,
        screen_container: document.getElementById("screen_container"),
        bios: {
            url: "./bios/seabios.bin",
        },
        vga_bios: {
            url: "./bios/vgabios.bin",
        },
        cdrom: {
            url: "./images/linux.iso",
        },
        autostart: true,
    });
}

window.onerror = function myErrorHandler(errorMsg, url, lineNumber) {
  document.getElementById('my_console').innerText += errorMsg + '\r\n';
  console.error(errorMsg, url, lineNumber);
}
</script>

<div id="screen_container">
    <div style="white-space: pre; font: 14px monospace; line-height: 14px"></div>
    <canvas style="display: none"></canvas>
</div>
```

**Comment by 1vers1on** [comment]
oh also im running this from an html file. (like with no web server)

**Reply by BelleNottelling** [reply]
> oh also im running this from an html file. (like with no web server)

Unless something has changed recently, v86 won't work without a webserver

**Reply by 1vers1on** [reply]
> > oh also im running this from an html file. (like with no web server)
> 
> Unless something has changed recently, v86 won't work without a webserver

Im pretty sure the built in python webserver is just an http sever to host the files

**Reply by randompersononinternet69** [reply]
Try this: npx http-server (install node.js if you haven't)

**Reply by SuperMaxusa** [reply]
https://github.com/copy/v86/blob/17a6b3b4e942ed04eb22098023d4b694f97020c0/Readme.md#L115-L117

Since v86 needs to load some files (v86.wasm, BIOS ROM and HDD/CD/FDD image) to work, you need an HTTP CORS-enabled file server, in fact, [`nginx`](https://nginx.org/), [`apache`](https://httpd.apache.org/), [`live-server`](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) and others may be suitable.

As written in the readme, the standard HTTP-server in Python 3 handles this well:

![2](https://user-images.githubusercontent.com/41739128/219848464-1e75c670-ca77-4a1e-adda-6db1de1ddbc8.png)

**Comment by Darin755** [comment]
If your trying to embed it on a different page you can use a iframe
